### PR TITLE
Novitsky/change database to my sql

### PR DESCRIPTION
**Pull request by SergeyNovitsky**
Change database from SQL Server to MySQL with Pomelo.EntityFrameworkCore
Update Microsoft.EntityFrameworkCore to version 5.0.11

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj [hunk 1]
```diff
@@ -23,15 +23,15 @@
...
+      <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="5.0.11" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.11" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="5.0.11">
         <PrivateAssets>all</PrivateAssets>
         <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       </PackageReference>
-      <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="5.0.2" />
-      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.2" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="5.0.11" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.11" />
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:34]: do we need SQL server dependency? MySQL maybe?

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj (hunk 1 above)
Comment (DmyMi) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:34]: p.s. can you add an example connection string to app settings for development?

**Review by Bogdan-Hasanov — APPROVED** [review]

**Review by toherberg — APPROVED** [review]

**Review by vladbond007 — APPROVED** [review]

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj (hunk 1 above)
Comment (SergeyNovitsky) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj]: fixed

File: OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20211107202949_InitialMigration.cs [hunk 2]
```diff
@@ -474,23 +573,23 @@ protected override void Up(MigrationBuilder migrationBuilder)
...
                 columns: table => new
                 {
-                    Id = table.Column<Guid>(type: "uniqueidentifier", nullable: false),
+                    Id = table.Column<byte[]>(type: "binary(16)", nullable: false),
                     Status = table.Column<int>(type: "int", nullable: false, defaultValue: 1),
-                    CreationTime = table.Column<DateTimeOffset>(type: "datetimeoffset", nullable: false),
-                    WorkshopId = table.Column<Guid>(type: "uniqueidentifier", nullable: false),
-                    ChildId = table.Column<Guid>(type: "uniqueidentifier", nullable: false),
-                    ParentId = table.Column<Guid>(type: "uniqueidentifier", nullable: false)
+                    CreationTime = table.Column<DateTimeOffset>(type: "datetime(6)", nullable: false),
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20211107202949_InitialMigration.cs:585]: one other thing that came to mind, can you check if we have time zone information (as we use `DateTimeOffset` in code) in database or that there is a correct conversion while writing/reading (e.g, we save `10:00+02:00` and it is stored as `08:00Z` and not as `10:00Z`)

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20211107202949_InitialMigration.cs (hunk 2 above)
Comment (SergeyNovitsky) [reply OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20211107202949_InitialMigration.cs:585]: yes, it converts time as UTC.

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj [hunk 3]
```diff
@@ -23,15 +23,15 @@
...
+      <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="5.0.11" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.11" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="5.0.11">
         <PrivateAssets>all</PrivateAssets>
         <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       </PackageReference>
-      <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="5.0.2" />
-      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.2" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="5.0.11" />
+      <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
```
[the remark this answers is not in the document]
Comment (SergeyNovitsky) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:34]: I will check it in other PR, where we will upgrade ours packages.

**Pull request merged**